### ISSUE TITLE
Incorrect MockEngine usage sample in documentation

### ISSUE THREAD

**Issue by Vantuz** [ktorio/ktor#1128]
### Ktor Version

1.2.0

### Ktor Engine Used(client or server and name)

MockEngine

### JVM Version, Operating System and Relevant Context

openjdk version "1.8.0_212"
Ubuntu 16.04

### Feedback

https://ktor.io/clients/http-client/testing.html#

I looked at the sample here and tried to compile following code:

```kotlin
package com.example

import io.ktor.client.HttpClient
import io.ktor.client.engine.mock.MockEngine
import io.ktor.client.engine.mock.respond
import io.ktor.http.*

class MockExample {
    val client = HttpClient(MockEngine) {
        engine {
            addHandler { request ->
                when (url.fullUrl) {
                    "https://example.org/" -> {
                        val responseHeaders = headersOf("Content-Type" to listOf(ContentType.Text.Plain.toString()))
                        respond("Hello, world", headers = responseHeaders)
                    }
                    else -> error("Unhandled ${url.fullUrl}")
                }
            }
        }
    }

    private val Url.hostWithPortIfRequired: String get() = if (port == protocol.defaultPort) host else hostWithPort
    private val Url.fullUrl: String get() = "${protocol.name}://$hostWithPortIfRequired$fullPath"
}
```
Got these errors:
```
e: /home/user/Projects/.../MockExample.kt: (12, 23): Unresolved reference: url
e: /home/user/Projects/.../MockExample.kt: (17, 48): Unresolved reference: url
```

I think the correct code should be this:
```kotlin
package com.example

import io.ktor.client.HttpClient
import io.ktor.client.engine.mock.MockEngine
import io.ktor.client.engine.mock.respond
import io.ktor.http.*

class MockExample {
    val client = HttpClient(MockEngine) {
        engine {
            addHandler { request ->
                when (request.url.fullUrl) { // changed here
                    "https://example.org/" -> {
                        val responseHeaders = headersOf("Content-Type" to listOf(ContentType.Text.Plain.toString()))
                        respond("Hello, world", headers = responseHeaders)
                    }
                    else -> error("Unhandled ${request.url.fullUrl}") // changed here
                }
            }
        }
    }

    private val Url.hostWithPortIfRequired: String get() = if (port == protocol.defaultPort) host else hostWithPort
    private val Url.fullUrl: String get() = "${protocol.name}://$hostWithPortIfRequired$fullPath"
}
```
But that way I get `Kotlin: This declaration is experimental and its usage must be marked with '@io.ktor.util.InternalAPI' or '@UseExperimental(io.ktor.util.InternalAPI::class)'` errors. Is there a way to access requested url (or other request data) without annotating my test class with `@InternalAPI`?

**Comment by e5l** [comment]
Fixed in `1.2.1`

**Comment by anykeyjunkie** [comment]
`@InternalAPI` annotation is not necessary any longer in 1.2.1, but documentation still needs the update as Vantuz pointed out
